### PR TITLE
refact(Kconfig): update as per lv_conf_template.h

### PR DESCRIPTION
**Pull request by rmshub**
### Description of the feature or fix
- Updated Kconfig as per latest lv_conf_template.h file.
- Removed unsupported options
- Rearranged few options
- Added new options

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by rmshub** [comment]
default configuration generated by Kconfiglib is
Config file: [sdkconfig.txt](https://github.com/lvgl/lvgl/files/14471738/sdkconfig.txt)
Header file: [sdkconfigheader.txt](https://github.com/lvgl/lvgl/files/14471743/sdkconfigheader.txt)

**Comment by rmshub** [comment]
Rebased to latest master branch to resolve the conflicts

**Review by kisvegabor — APPROVED** [review]
Looks good to me, although I'm not using Kconfig. @FASTSHIFT, @XuNeo can you try it out?

**Comment by FASTSHIFT** [comment]
> Looks good to me, although I'm not using Kconfig. @FASTSHIFT, @XuNeo can you try it out?

I tested this PR on my hardware platform and it works fine :)

File: Kconfig [hunk 1]
```diff
@@ -1170,18 +1169,18 @@ menu "LVGL configuration"
 	menu "Others"
 		config LV_USE_SNAPSHOT
 			bool "Enable API to take snapshot"
-			default y if !LV_CONF_MINIMAL
+			default n if !LV_CONF_MINIMAL
```
Comment (FASTSHIFT) [inline Kconfig:1172]: Why turn off by default?

Comment (rmshub) [reply Kconfig:1172]: Turned off by default based on https://github.com/lvgl/lvgl/blob/master/lv_conf_template.h#L746. Do we need Snapshot functionality ON by default?

Comment (FASTSHIFT) [reply Kconfig:1172]: OK, let's be consistent with lv_conf_template.

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**